src/test/scala/edu/ie3/simona/agent/grid/PowerFlowSupportSpec.scala

```diff
@@ -366,4 +387,143 @@ class PowerFlowSupportSpec
+        )
+
+        val pf = newtonRaphsonPF(
+          gridModel,
+          50,
+          operationPoint,
+          slackNodeVoltages,
+        )(Vector(1e-12)) match {
+          case result: PowerFlowResult.SuccessFullPowerFlowResult =>
+            result
+          case _: PowerFlowResult.FailedPowerFlowResult =>
+            fail()
+        }
+
+        val indexMap = gridModel.nodeUuidToIndexMap.map { case (uuid, i) =>
+          (i, uuid)
+        }
+        pf.nodeData.map(n => indexMap(n.index) -> Each(n.voltage.abs)).toMap
+      }
+
+      val onlyLvPowerFlowResults: Map[UUID, Dimensionless] = {
+        val (gridModel, receivedValueStore) = TestData.onlyLv
+
+        val pf = slackGridPF(
+          gridModel,
+          receivedValueStore,
+          powerFlowParams,
+        ) match {
+          case result: PowerFlowResult.SuccessFullPowerFlowResult =>
+            result
+          case _: PowerFlowResult.FailedPowerFlowResult =>
+            fail()
+        }
+
+        val indexMap = gridModel.nodeUuidToIndexMap.map { case (uuid, i) =>
+          (i, uuid)
+        }
+        pf.nodeData.map(n => indexMap(n.index) -> Each(n.voltage.abs)).toMap
+
+      }
+
+      onlyLvPowerFlowResults.foreach { case (uuid, result) =>
+        withMvPowerFlowResults(uuid) should approximate(result)(tolerance)
+      }
+    }
+  }
+
+  object TestData extends DbfsTestGrid with ConfigTestData {
+    val time: Simona.Time = simonaConfig.simona.time
+
+    implicit def toZoneDateTime(time: String): ZonedDateTime =
+      TimeUtil.withDefaults.toZonedDateTime(time)
+
+    implicit def toGridModel(
+        subGridContainer: SubGridContainer
+    ): GridModel =
+      GridModel(
+        subGridContainer,
+        RefSystem("2000 MVA", "110 kV"),
+        time.startDateTime,
+        time.endDateTime,
+        simonaConfig,
+      )
+
+    val withMv: (GridModel, ReceivedValuesStore) = {
+      val gridModel = TestGridFactory.createSubGrid(
+        gridName = "centerGrid",
+        subgrid = 1,
+        rawGridElements = new RawGridElements(
+          Set(supNodeA, node1, node2, node3, node4).asJava,
+          Set(line1To2, line2To3, line1To4).asJava,
+          Set(transformer1).asJava,
+          Set.empty[Transformer3WInput].asJava,
+          Set.empty[SwitchInput].asJava,
+          Set.empty[MeasurementUnitInput].asJava,
+        ),
+      )
+
+      val receivedValueStore = ReceivedValuesStore.empty(
+        Map.empty,
+        Map.empty,
+        Vector(supNodeA.getUuid),
+      )
+
+      (gridModel, receivedValueStore)
+    }
+
+    val onlyLv: (GridModel, ReceivedValuesStore) = {
+      val updatedNode1 =
+        node1.copy().slack(true).vTarget(0.9999984268502677.asPu).build()
+      val gridModel = TestGridFactory.createSubGrid(
+        gridName = "centerGrid",
+        subgrid = 1,
+        rawGridElements = new RawGridElements(
+          Set(updatedNode1, node2, node3, node4).asJava,
```

Comment:
These nodes are HV. To avoid later confusion, maybe we should use matching nodes here?

Reply:
Same for MV above